src/expression/binary.cpp

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Team Dissolve and contributors
+
+#include "expression/binary.h"
+
+ExpressionBinaryOperatorNode::ExpressionBinaryOperatorNode(BinaryOperator op) : ExpressionNode(), operator_(op) {}
+
+/*
+ * Evaluation
+ */
+
+// Evaluate node
+std::optional<ExpressionValue> ExpressionBinaryOperatorNode::evaluate() const
+{
+    // Must be two child nodes
+    if (children_.size() != 2)
+        return std::nullopt;
+
+    // Evaluate LHS and RHS nodes
+    auto optl = children_[0]->evaluate();
+    if (!optl)
+        return std::nullopt;
+    auto optr = children_[1]->evaluate();
+    if (!optr)
+        return std::nullopt;
+
+    // Perform the operation
+    ExpressionValue result;
+    auto lhs = (*optl), rhs = (*optr);
+    switch (operator_)
+    {
+        case (OperatorAdd):
+            if (ExpressionValue::bothIntegers(lhs, rhs))
+                result = lhs.asInteger() + rhs.asInteger();
+            else
+                result = lhs.asDouble() + rhs.asDouble();
+            break;
+        case (OperatorDivide):
+            if (ExpressionValue::bothIntegers(lhs, rhs))
+                result = lhs.asInteger() / rhs.asInteger();
+            else
+                result = lhs.asDouble() / rhs.asDouble();
+            break;
+        case (OperatorSubtract):
+            if (ExpressionValue::bothIntegers(lhs, rhs))
+                result = lhs.asInteger() - rhs.asInteger();
+            else
+                result = lhs.asDouble() - rhs.asDouble();
+            break;
+        case (OperatorPow):
+            result = pow(lhs.asDouble(), rhs.asDouble());
+            break;
+        case (OperatorMultiply):
+            if (ExpressionValue::bothIntegers(lhs, rhs))
+                result = lhs.asInteger() * rhs.asInteger();
+            else
+                result = lhs.asDouble() * rhs.asDouble();
+            break;
+    }
```

Comment:
```suggestion
    typeof(std::plus) op;
    switch (operator_)
    {
        case (OperatorAdd):  op = std::plus; break;
        case (OperatorSubtract):  op = std::minus; break;
        case (OperatorDivide):op = std::divides; break
        case (OperatorPow):  op = pow; break;
        case (OperatorMultiply):  op = std::multiply;
    }
    
    if (ExpressionValue::bothIntegers(lhs, rhs))
        result = op(lhs.asInteger(), rhs.asInteger());
    else
        result = op(lhs.asDouble(), rhs.asDouble());
```

I /believe/ we can do this.  It'll also require including <functional> at the beginning.

Reply:
Nice idea, however, it would appear that these are templated functions, and so need the argument types in the declaration, which kinda defeats the purpose!